library/cartridge_edit_topology.py

```diff
@@ -407,51 +454,80 @@ def get_servers_params(instances, cluster_instances, allow_missed_instances):
     return servers_params, None
 
 
-def get_topology_params(replicasets, cluster_replicasets, instances, cluster_instances, allow_missed_instances):
-    topology_params = {}
+##############
+# Check mode #
+##############
 
-    replicasets_params, err = get_replicasets_params(
-        replicasets, cluster_replicasets, cluster_instances, allow_missed_instances
-    )
-    if err is not None:
-        return None, err
+def check_instances_for_dangerous_changes(all_instances, instances, cluster_instances, ignore_checks):
+    errors = []
+    extra_cluster_instances = []
+    changed_advertise_uris = []
 
-    if replicasets_params:
-        topology_params['replicasets'] = replicasets_params
+    for instance_name in cluster_instances.keys():
+        if instance_name not in all_instances:
+            extra_cluster_instances.append(instance_name)
+        elif instance_name in instances:
+            uri = instances[instance_name].get('uri')
+            cluster_uri = cluster_instances[instance_name].get('uri')
+            if uri != cluster_uri:
+                changed_advertise_uris.append('%s (%s -> %s)' % (instance_name, cluster_uri, uri))
 
-    servers_params, err = get_servers_params(instances, cluster_instances, allow_missed_instances)
-    if err is not None:
-        return None, err
+    if extra_cluster_instances and not ignore_checks.get(EXTRA_CLUSTER_INSTANCES_CHECK_NAME):
+        msg = 'Some instances from cluster are missing in inventory: %s' % ', '.join(extra_cluster_instances)
+        errors.append(msg)
 
-    if servers_params:
-        topology_params['servers'] = servers_params
+    if changed_advertise_uris and not ignore_checks.get(ADVERTISE_URIS_CHANGE_CHECK_NAME):
+        msg = 'Advertise uris of some instances changed: %s' % ', '.join(changed_advertise_uris)
+        errors.append(msg)
 
-    return topology_params, None
+    return errors
 
 
-def get_replicasets_failover_priority_and_instances_params(
-    replicasets, cluster_replicasets, instances, cluster_instances, allow_missed_instances
-):
-    topology_params = {}
+def check_replicasets_for_dangerous_changes(all_replicasets, replicasets, cluster_replicasets, ignore_checks):
+    errors = []
+    extra_cluster_replicasets = []
+    renamed_replicasets = []
 
-    replicasets_params, err = get_replicasets_params_for_changing_failover_priority(
-        replicasets, cluster_replicasets, cluster_instances, allow_missed_instances
-    )
-    if err is not None:
-        return None, err
+    for cluster_replicaset_name, cluster_replicaset_params in cluster_replicasets.items():
+        if cluster_replicaset_name not in all_replicasets:
+            renamed = False
 
-    if replicasets_params:
-        topology_params['replicasets'] = replicasets_params
+            for replicaset_name, replicaset_params in all_replicasets.items():
```

Comment:
I suggest checking `replicaset_alias` for each instance:

* `cluster_instance[replicaset_alias] == instance[replicaset_alias]` - OK
* `cluster_instance[replicaset_alias] == nil` - OK
* `cluster_instance[replicaset_alias] != instance[replicaset_alias]` - BAD: replicaset was renamed or instance was moved to the other one.

Generally, I'm not sure we should check it. Currently, replicaset renaming isn't supported since the alias is a replicaset identifier. It is not a "dangerous" change, it's impossible. So, it can be non-skipped error on `edit_topology` while we don't allow to rename replicasets.

Reply:
We haven't replicaset_alias for cluster instances

Reply:
Also, there are many situations when checking a replica set through INSTANCE will give false-positive results